### DISCUSSION TITLE
32-bit version.

### DISCUSSION THREAD

**Discussion by Mulligan81** [wingtk/gvsbuild#1386] [General]
Is there a 32-bit pre-build version of gvsbuild?

**Comment by danyeaw** [comment]
Hi @Mulligan81, for which operating system?

**Reply by sreekanthbatthula** [reply]
Hello @danyeaw  
I have the same requirement for GTK 4 32-bit version for Win 10 & Win 8.1. Could you please let me know?

**Reply by Mulligan81** [reply]
> Hi @Mulligan81, for which operating system?

Sorry, haven't mentioned, Windows 10.

**Reply by Mulligan81** [reply]
> Hello @Mulligan81, I have the same requirement for GTK 4 32-bit version for Win 10 & Win 8.1. Could you please let me know?
So now there is at least 2 people with the same request;-)

**Comment by danyeaw** [comment]
This is technically possible by us building everything in a second build with the platform set to x86. I am kind of torn about doing this though. Market share of 32-bit Windows is much less than 1%. If we want to build final binaries using x86, it also means we should probably test the build on 32-bit also, which would double the number of builds each time our CI/CD pipeline is run. Each build takes significant CPU cycles to build all of GTK3 and GTK4. It would also be another thing we need to support.

If someone is really excited about 32-bit builds, I would feel better about it if someone wanted to help support them.

**Reply by sreekanthbatthula** [reply]
Hello @danyeaw ,
Thanks for your prompt response. I understand your point. Is there a possibility you can share a script (that was verified) we can use to build 32-bit binaries?
I tried steps from https://www.gtk.org/docs/installations/windows/. But neither of them was successful.

If this is big ask, then can you point me to archives from where I can get 32-bit and 64-bit binaries of GTK 3 (THREE).?

**Reply by danyeaw** [reply]
https://github.com/wingtk/gvsbuild, go to the Build GTK section of the README.

**Reply by sreekanthbatthula** [reply]
I already tried those steps and landed up with errors around glib etc. 
Do you know where can I find 32-bit and 64-bit archived sources for GKT 3? At least for the Latest old stable 3.24.43.

**Reply by danyeaw** [reply]
I don't know of any place that has current GTK 3 binaries. If you are running into errors with Gvsbuild, please submit an issue with the details of your setup, what commands you used, and a traceback with the GLib compile errors.

**Reply by hsriperu** [reply]
Sorry to hijack the thread with some unrelated question but I am a bit desperate here. I work with @sreekanthbatthula and he has high regards for your GTK knowledge. 

I have a question with GTK4 window sizing that I posted here - https://discourse.gnome.org/t/how-to-build-fixed-size-windows-in-gtk-4/22807. @danyeaw, If you got any pointers, I would highly appreciate it. Thank you !
